### PR TITLE
docs: Improve README

### PR DESCRIPTION
**Pull request by akcelero**

### CODE REVIEW

**Review by mokrzesa — APPROVED** [review]
WOW - IMPRESSIVE WORK

**Review by mokrzesa — APPROVED** [review]
WOW - IMPRESSIVE WORK

File: README.md [hunk 1]
```diff
@@ -1,2 +1,47 @@
...
+The proofs are the sibling values collected on this path.
+
+**Note**
+- To find a sibling's node number, use `current_node_number ^ 1`.
+- To find a parent's node number, use `current_node_number // 2`.
+
+### Verifying Proofs
+
+To verify proofs, you only need the root value. For example:
+```
```
Comment (sergii-denysiuk) [inline README.md:39]: nit:
```suggestion
```python
```

**Review by sergii-denysiuk — APPROVED** [review]

File: README.md [hunk 2]
```diff
@@ -1,2 +1,47 @@
...
+value = b
+proofs = [A, E, CD]
+```
+1. Hash the value `b` to get `B`.
+2. Combine `B` with the first proof and hash result: `hash(A + B) = AB`.
+3. Combine `AB` with the second proof and hash result: `hash(AB + E) = ABE`.
+4. Combine `ABE` with the third proof and hash result: `hash(ABE + CD) = ABCDE`, which is the *calculated root*.
```
Comment (sergii-denysiuk) [inline README.md:46]:
```suggestion
4. Combine `ABE` with the third proof and hash result: `hash(ABE + CD) = ABECD`, which is the *calculated root*.
```

Comment (akcelero) [reply README.md:46]: thanks!

**Review by sergii-denysiuk — APPROVED** [review]
Looks great, and very understandable.

**Pull request merged**